packages/mobile/src/verify/VerificationFailedModal.tsx

```diff
@@ -3,34 +3,54 @@ import colors from '@celo/react-components/styles/colors'
 import fontStyles from '@celo/react-components/styles/fonts'
 import { componentStyles } from '@celo/react-components/styles/styles'
 import * as React from 'react'
-import { WithTranslation } from 'react-i18next'
+import { useTranslation } from 'react-i18next'
 import { StyleSheet, Text, View } from 'react-native'
 import Modal from 'react-native-modal'
-import { Namespaces, withTranslation } from 'src/i18n'
+import { Namespaces } from 'src/i18n'
+import { cancelVerification } from 'src/identity/actions'
+import { VerificationStatus } from 'src/identity/verification'
 import { navigateHome } from 'src/navigator/NavigationService'
 
-interface OwnProps {
-  isVisible: boolean
+interface Props {
+  verificationStatus: VerificationStatus
+  cancelVerification: typeof cancelVerification
 }
 
-type Props = OwnProps & WithTranslation
+export function VerificationFailedModal(props: Props) {
+  const { t } = useTranslation(Namespaces.nuxVerification2)
+  const [isDismissed, setIsDismissed] = React.useState(false)
+
+  const onDismiss = () => {
```

Comment:
As far as I understand, that would create new lambda every time something changes in props, right? Shall we use `useCallback` here?

Reply:
Done :)